src/compiler/crystal/tools/doc/item.cr

```diff
@@ -6,4 +6,68 @@ module Crystal::Doc::Item
   def formatted_summary
     @generator.summary(self)
   end
+
+  macro def_to_json(type, properties)
+    Item.def_to_json(nil, {{properties}})
```

Comment:
I explained the purpose in the initial post. This is basically the to-json part of `JSON.mapping`. It would probably be better to move this to the `JSON` namespace because it is generally useful to easily define a `to_json` method with mappings as data if you don't want the full features of `JSON.mapping`.